flow-model-generator/modules/flow-model-generator-core/src/main/java/io/ballerina/flowmodelgenerator/core/type/RecordValueGenerator.java

```diff
@@ -0,0 +1,212 @@
+                                    fieldValues.add(fieldObj.get("name").getAsString() + ": " + sb);
+                                }
+                                case "union" -> {
+                                    StringBuilder sb = new StringBuilder();
+                                    generateUnionValue(fieldObj, sb);
+                                    fieldValues.add(fieldObj.get("name").getAsString() + ": " + sb);
+                                }
+                                default -> {
+                                    if (fieldObj.has("value") && !fieldObj.get("value")
+                                            .getAsString().isEmpty()) {
+                                        fieldValues.add(fieldObj.get("name").getAsString()
+                                                + ": " + fieldObj.get("value").getAsString());
+                                    } else if (fieldObj.has("defaultValue")
+                                            && !fieldObj.get("defaultValue").getAsString().isEmpty()) {
+                                        fieldValues.add(fieldObj.get("name").getAsString() +
+                                                ": " + fieldObj.get("defaultValue").getAsString());
+                                    } else {
+                                        StringBuilder sb = new StringBuilder();
+                                        generateDefaultValue(fieldObj, sb);
+                                        fieldValues.add(fieldObj.get("name").getAsString() + ": " + sb);
+                                    }
+                                }
+                            }
+                        }
+                    }
+                }
+            }
+        }
+        builder.append(String.join(", ", fieldValues));
+        builder.append("}");
+    }
+
+    private static void generateDefaultValue(JsonObject jsonObject, StringBuilder builder) {
+        if (!jsonObject.has("typeName")) {
+            return;
+        }
+        String typeName = jsonObject.get("typeName").getAsString();
+        switch (typeName) {
+            case "record" -> generateRecordValue(jsonObject, builder);
+            case "union" -> generateUnionValue(jsonObject, builder);
+            case "array" -> builder.append("[]");
+            case "enum" -> {
+                if (jsonObject.has("members") && jsonObject.get("members").isJsonArray()) {
+                    JsonElement members = jsonObject.get("members");
+                    if (!members.getAsJsonArray().isEmpty()) {
+                        generateDefaultValue(members.getAsJsonArray().get(0).getAsJsonObject(), builder);
+                    } else {
+                        builder.append("\"\"");
+                    }
+                }
+            }
+            case "error" -> builder.append("error(\"Custom Error\")");
+            case "map" -> builder.append("{}");
+            case "object" -> builder.append("object {}");
```

Comment:
This experience is not ideal; we must also check the object type, like whether it is a raw template compatible type, then the default value should be ` `` ` , else if it is class then `new()` etc.

Reply:
I will check an update in the next PR.